core/src/main/java/org/apache/calcite/sql/validate/SqlValidatorImpl.java

```diff
@@ -2256,11 +2256,16 @@ protected void registerNamespace(
       @Nullable String alias,
       SqlValidatorNamespace ns,
       boolean forceNullable) {
-    namespaces.put(requireNonNull(ns.getNode(), () -> "ns.getNode() for " + ns), ns);
+    final SqlNode sqlNode = requireNonNull(ns.getNode(), () -> "ns.getNode() for " + ns);
```

Comment:
Consider removing the `sqlNode` variable. After you've called `requireNonNull(ns.getNode())` the validator will no longer complain.

Reply:
Removed `sqlNode`.